File: c/gpt/p7.c
// Task p7: atoi-like conversion of argv[1] and print argv[INDEX]
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char **argv)
{
    if (argc < 2)
    {
        printf("Usage: %s INDEX [args...]\n", argv[0]);
        return 0;
    }

    int idx = atoi(argv[1]);
    if (idx >= 0 && idx < argc)
    {
        printf("%s\n", argv[idx]);
    }
    else
    {
        printf("\n");
    }

    return 0;
}

File: python/gpt/p7.py
# Task p7: Use atoi-like conversion on command-line argv[1] to pick which argv to print
import sys


def main():
    # Expect at least two arguments: program and index
    if len(sys.argv) < 2:
        print("Usage: python p7.py INDEX [args...]")
        return 0

    try:
        index = int(sys.argv[1])
    except ValueError:
        print("Invalid INDEX")
        return 0

    target = index
    # We interpret the spec as printing argv[target] where argv[0] is program,
    # so ensure target is within range of sys.argv
    if 0 <= target < len(sys.argv):
        print(sys.argv[target])
    else:
        print("")

    return 0


if __name__ == "__main__":
    sys.exit(main())
